lib/main.coffee

```diff
@@ -0,0 +1,44 @@
+{CompositeDisposable} = require 'atom'
+
+module.exports =
+  config:
+    executablePath:
+      type: 'string'
+      title: 'PHP Executable Path'
+      default: 'php' # Let OS's $PATH handle the rest
+
+  activate: ->
+    @regex = '(Parse|Fatal) ' +
+            '(?<error>error):(\\s*(?<type>parse|syntax) error,?)?\\s*' +
+            '(?<message>(unexpected \'(?<near>[^\']+)\')?.*) ' +
+            'in .*? on line (?<line>\\d+)'
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-php.executablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    path = require 'path'
+    helpers = require('atom-linter')
+    child_process = require 'child_process'
+    provider =
+      grammarScopes: ['text.html.php', 'source.php']
+      scope: 'file'
+      lintOnFly: false
+      lint: (textEditor) =>
+        return new Promise (resolve, reject) =>
```

Comment:
How about we don't return a promise here, and return `helpers.exec` return value and add a then to it that alters it's value?